tests/manage/noobaa/conftest.py

```diff
@@ -0,0 +1,109 @@
+import logging
+
+import pytest
+
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.resources import noobaa
+from tests import helpers
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture()
+def noobaa_obj():
+    """
+    Returns a NooBaa resource that's connected to the S3 endpoint
+    Returns:
+        noobaa_obj: A NooBaa resource
+
+    """
+
+    noobaa_obj = noobaa.NooBaa()
+    return noobaa_obj
+
+
+@pytest.fixture()
+def uploaded_objects(request, noobaa_obj, awscli_pod):
+
+    """
+    Deletes all objects that were created as part of the test
+
+    Returns:
+        An empty list of objects
+
+    """
+
+    uploaded_objects = []
+
+    base_command = f"sh -c \"AWS_ACCESS_KEY_ID={noobaa_obj.access_key_id} " \
+        f"AWS_SECRET_ACCESS_KEY={noobaa_obj.access_key} " \
+        f"AWS_DEFAULT_REGION=us-east-1 " \
+        f"aws s3 " \
+        f"--endpoint={noobaa_obj.endpoint} "
+
+    def object_cleanup():
+        for uploaded_filename in uploaded_objects:
+            # TODO: Add assert
+            logger.info(f'Deleting object {uploaded_filename}')
+            awscli_pod.exec_cmd_on_pod(
+                command=base_command + "rm " + uploaded_filename + "\""
+            )
+    request.addfinalizer(object_cleanup)
+    return uploaded_objects
+
+
+@pytest.fixture()
+def created_buckets(request, noobaa_obj):
+    """
+    Deletes all buckets that were created as part of the test
+
+    Returns:
+        An empty list of buckets
+
+    """
+
+    created_buckets = []
+
+    def bucket_cleanup():
+        for bucket in created_buckets:
+            logger.info(f'Deleting bucket {bucket.name}')
+            bucket.object_versions.delete()
+            noobaa_obj.s3_delete_bucket(bucket)
+    request.addfinalizer(bucket_cleanup)
+    return created_buckets
+
+
+@pytest.fixture()
+def created_pods(request):
+    """
+    Deletes all pods that were created as part of the test
+
+    Returns:
+        An empty list of pods
+
+    """
+
```

Comment:
delete empty line

Reply:
If I'm not wrong, our styling guide is to keep an empty line after the method doc, no?

Reply:
no need AFAIK

Reply:
I don't think we actually specify on this. We do want an empty line at the end of the doc, however we don't have any rule for the first line after the doc. That being said I primarily see no empty line after the doc in our codebase so we should probably stick with that convention.

Reply:
Thanks all, fixed :+1: